src/models/appmodel.ts

```diff
@@ -25,7 +30,10 @@ export class AppModel implements IAppModel {
     };
     
     @action addFlavor = () => {
+        // PROBLEM: These actions should cause the combobox to rerender
+        // automatically, but that does not happen
         this.flavors.push(this.flavorInput);
```

Comment:
An observer component will re-render once the observables that got de-referenced in the previous render are changed. The push does not de-reference your contacts array. It would work if you wrote slice() or listen to changes via observe https://mobx.js.org/refguide/array.html

or just like I always do, create a new array and assign it instead of the old one. Thus you'll receive immutable structure

Reply:
It appears that updating the array already works as expected, now that I think about it.  The only remaining item is to make it update automatically when selectedFlavor changes on the appModel.   I updated the card on MistyMountains for this work.